### ISSUE TITLE
The argument --keep-going does not seem to be working. I can't seem to get past one of the crates having a 403 forbidden Error.

### ISSUE THREAD

**Issue by f34rt3hbunn3h** [drmikehenry/romt#8]
I am attempting to mirror crates for offline use, but I can't seem to get past downloading certain crates when there is a 403 forbidden. The only way for me to get past the affected crates is to create an empty file and do a download with --assume-ok option.

On wsl2 Ubuntu 20.04 from pip install romt:
```bash
romt crate -v --keep-going init export
...
[downloading] crates/bo/rk/bork/bork-0.0.0.crate
Traceback (most recent call last):
  File "/home/user/.local/bin/romt", line 8, in <module>
    sys.exit(main())
  File "/home/user/.local/lib/python3.8/site-packages/romt/cli.py", line 172, in main
    romt.crate.Main(args).run()
  File "/home/user/.local/lib/python3.8/site-packages/romt/base.py", line 50, in run
    self._run()
  File "/home/user/.local/lib/python3.8/site-packages/romt/crate.py", line 916, in _run
    getattr(self, cmd_func_name)()
  File "/home/user/.local/lib/python3.8/site-packages/romt/crate.py", line 831, in cmd_download
    download_crates(
  File "/home/user/.local/lib/python3.8/site-packages/romt/crate.py", line 341, in download_crates
    _process_crates(
  File "/home/user/.local/lib/python3.8/site-packages/romt/crate.py", line 317, in _process_crates
    downloader.run_job(_process_inner)
  File "/home/user/.local/lib/python3.8/site-packages/romt/download.py", line 223, in run_job
    trio.run(functools.partial(job, **kwargs), *args)
  File "/home/user/.local/lib/python3.8/site-packages/trio/_core/_run.py", line 1932, in run
    raise runner.main_task_outcome.error
  File "/home/user/.local/lib/python3.8/site-packages/romt/crate.py", line 313, in _process_inner
    nursery.start_soon(
  File "/home/user/.local/lib/python3.8/site-packages/trio/_core/_run.py", line 815, in __aexit__
    raise combined_error_from_nursery
  File "/home/user/.local/lib/python3.8/site-packages/romt/crate.py", line 285, in _process_one
    await downloader.adownload_verify_hash(
  File "/home/user/.local/lib/python3.8/site-packages/romt/download.py", line 129, in adownload_verify_hash
    await self.adownload(dest_url, dest_path)
  File "/home/user/.local/lib/python3.8/site-packages/romt/download.py", line 58, in adownload
    await self._adownload(url, tmp_dest_path)
  File "/home/user/.local/lib/python3.8/site-packages/romt/download.py", line 48, in _adownload
    await self.adownload_fileobj(url, f)
  File "/home/user/.local/lib/python3.8/site-packages/romt/download.py", line 35, in adownload_fileobj
    response.raise_for_status()
  File "/home/user/.local/lib/python3.8/site-packages/httpx/_models.py", line 1402, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: 403 Client Error: Forbidden for url: https://static.crates.io/crates/bork/bork-0.0.0.crate
For more information check: https://httpstatuses.com/403
```

Then I tried the following after downloading the stand alone executable:
```bash
./romt-0.3.0-x86_64-linux crate --keep-going --num-jobs 50 download
download...
Traceback (most recent call last):
  File "romt-wrapper.py", line 5, in <module>
  File "romt/cli.py", line 172, in main
  File "romt/base.py", line 50, in run
  File "romt/crate.py", line 916, in _run
  File "romt/crate.py", line 839, in cmd_download
  File "romt/crate.py", line 349, in download_crates
  File "romt/crate.py", line 317, in _process_crates
  File "romt/download.py", line 223, in run_job
  File "trio/_core/_run.py", line 1932, in run
  File "romt/crate.py", line 314, in _process_inner
  File "trio/_core/_run.py", line 815, in __aexit__
  File "romt/crate.py", line 286, in _process_one
  File "romt/download.py", line 129, in adownload_verify_hash
  File "romt/download.py", line 58, in adownload
  File "romt/download.py", line 48, in _adownload
  File "romt/download.py", line 35, in adownload_fileobj
  File "httpx/_models.py", line 1402, in raise_for_status
httpx.HTTPStatusError: 403 Client Error: Forbidden for url: https://static.crates.io/crates/bork/bork-0.2.0.crate
For more information check: https://httpstatuses.com/403
[1920] Failed to execute script 'romt-wrapper' due to unhandled exception!
```

**Comment by f34rt3hbunn3h** [comment]
In testing adding httpx.HTTPStatusError to the try, except in download.py: Class Downloader: function adownload_fileobj, seeems to fix the issue. Currently testing to see if there is any problems before making a pull request.

**Comment by drmikehenry** [comment]
It seems I incorrectly switched from the `requests` library's `RequestException` base class to the `httpx` library's `RequestError` class; the former is the base class for all of the exceptions in `requests`, whereas the latter doesn't cover all exceptions in `httpx`.  It looks like I should have used `httpx.HTTPError` instead of `httpx.RequestError` as the catch-all type.

I'd be happy to have a pull request catching `httpx.HTTPError` instead of `httpx.RequestError`.  Thanks for finding this problem and tracking down the solution :-)

References:
- https://docs.python-requests.org/en/master/_modules/requests/exceptions/
- https://www.python-httpx.org/exceptions/

**Comment by drmikehenry** [comment]
Thanks! the fix has been released in v0.3.1.
